foreman/data_refinery_foreman/surveyor/harmony.py

```diff
@@ -27,11 +27,10 @@ def extract_title(sample: Dict) -> str:
         "extract name",
     ]
     title_fields = add_variants(title_fields)
-    for key, value in sorted(sample.items(), key=lambda x: x[0].lower()):
-        lower_key = key.lower().strip()
 
-        if lower_key in title_fields:
-            return value
+    for title_field in title_fields:
+        if title_field in sample:
+            return sample[title_field]
```

Comment:
I'm not actually sure if I should leave this change in or not. I think just using `harmony.extract_title()` in both places will be enough to fix it. I do think that preferring `title` over `sample name` would be a beneficial change, but this also is less consistent with how we've done it previously.

I'm leaning with changing it rather than staying consistent, but I'd love to hear other opinions if they exist.

Reply:
I think that the code you have now is better, regardless of whether or not we want to keep preferring `title` over `sample name`. Looking at the old code, I couldn't tell why `sample name` had higher priority than `title` until I realized that it would sort earlier. No matter what we choose, I think you should add a comment above `title_fields` that says something like "sorted in decreasing order of priority".

I am hesitant to change the default behavior though, because a lot of our tests are written using the implicit assumption that `sample name` is prioritized over `title` since they take metadata from prod rather than surveying every time.

Reply:
We also use `title` in places like column detection in the Illumina processor so changing the behavior has a nontrivial effect on our processing.